markov_chain.py

```diff
@@ -0,0 +1,121 @@
+        >>> a = Markov(open('test.txt'))
+        >>> a.words
+        ['This', 'is', 'a', 'test.']
+        >>> a = Markov(open('test.txt'))
+        >>> a.word_size
+        4
+        """
+        return 'Markov({!r}, {!r}, {!r})'.format(self.cache, self.words,
+                                                 self.word_size)
+
+    def __eq__(self, other):
+        r"""Defines equality in Markov Objects.
+
+        >>> a = Markov(open('test.txt'))
+        >>> b = Markov(open('test.txt'))
+        >>> a == b
+        True
+        >>> a = Markov(open('test.txt'))
+        >>> b = Markov(open('not_test.txt'))
+        >>> a == b
+        False
+        """
+        return (self.cache == other.cache and self.words == other.words and
+                self.word_size == other.word_size)
+
+    def file_to_words(self):
+        """Opens a text file and splits all the words into a list.
+
+        >>> a = Markov(open('test.txt'))
+        >>> a.file_to_words()
+        ['This', 'is', 'a', 'test.']
+        """
+        self.open_file.seek(0)
+        data = self.open_file.read()
+        words = data.split()
+        return words
+
+    def triples(self):
+        """Generates triples (3-grams) from the given data string.
+
+        >>> a = Markov(open('test.txt'))
+        >>> b = a.triples()
+        >>> list(b)
+        [('This', 'is', 'a'), ('is', 'a', 'test.')]
+        """
+
+        if len(self.words) < 3:
+            return
+
+        for i in range(len(self.words) - 2):
+            yield (self.words[i], self.words[i + 1], self.words[i + 2])
+
+    def database(self):
+        """Creates a data base of words from triples and number of words.
+        """
+        for w1, w2, w3 in self.triples():
+            key = (w1, w2)
+            if key in self.cache:
+                self.cache[key].append(w3)
+            else:
+                self.cache[key] = [w3]
+
+    def generate_top_text(self, size=(random.randrange(1, 7))):
+        """Generates a phrase from the sample text anywhere between 1 and 6 tokens long."""
+        seed = random.randint(0, self.word_size - 3)
+        seed_word, next_word = self.words[seed], self.words[seed + 1]
+        w1, w2 = seed_word, next_word
+        gen_words = []
+        for i in range(size):
+            gen_words.append(w1)
+            w1, w2 = w2, random.choice(self.cache[(w1, w2)])
+        gen_words.append(w2)
+        return ' '.join(gen_words)
+
+    def generate_bottom_text(self, size=(random.randrange(1, 11))):
+        """Generates a phrase from the sample text anywhere between 1 and 10 tokens long."""
+        seed = random.randint(0, self.word_size - 3)
+        seed_word, next_word = self.words[seed], self.words[seed + 1]
+        w1, w2 = seed_word, next_word
+        gen_words = []
+        for i in range(size):
+            gen_words.append(w1)
+            w1, w2 = w2, random.choice(self.cache[(w1, w2)])
+        gen_words.append(w2)
+        return ' '.join(gen_words)
```

Comment:
Both of these functions are identical except for the default argument for `size`. I would not duplicate this code and instead in your `text_generator` module, have the calling functions set those parameters. Then this module just deals with Markov in the abstract and your text generator module is specifically for memes.